src/libutil/file-system.cc

```diff
@@ -221,7 +221,11 @@ bool pathAccessible(const Path & path)
 Path readLink(const Path & path)
 {
     checkInterrupt();
-    return fs::read_symlink(path).string();
+    try {
+        return fs::read_symlink(path).string();
+    } catch (fs::filesystem_error & e) {
+        throw SysError("reading symlink: %1%", e.what());
```

Comment:
This depends on errno. I will probably skip it for now.

Reply:
What do you mean "depends on errno"? We can get those out of `filesystem_error` pretty easily, actually.

Reply:
Seems to work.